### DISCUSSION TITLE
Amiberry not remembering ROM files

### DISCUSSION THREAD

**Discussion by jameswhit82** [BlitterStudio/amiberry#1479] [Q&A] [answered]
Hi,

I’ve got a a strange issue where Amiberry is not remembering my Amiga ROM files. These are ones that I took from Amiga Forever and include a Rom.key file.

Every time I reload I have to re select the appropriate ROM.

Any advice would be much appreciated!

Thanks you!

**Comment by midwan** [comment]
What do you mean by "not remembering"?
Are they not listed in the dropdown?
Are you loading a config with a specific ROM, and that doesn't work?
Please provide more details on what you're trying to do exactly, including the version of Amiberry you're testing with.

**Reply by jameswhit82** [reply]
Hi. Apologies, I should have mentioned that in my previous post. This is Amiberry v5.7.4 running on Mac OS 15.0.1.

I get nothing listed under ‘Main ROM file’ other than the AROS built in ROM. 

![image](https://github.com/user-attachments/assets/429a1013-c1a0-4d1a-bbd6-7d4c969a73fe)

Although I have all of the below from Amiga Forever:

![image](https://github.com/user-attachments/assets/b526a769-e60e-4a33-8e87-2f02fd90a5dc)

It seems to run happily, it’s just annoying every time I load my configuration I have to manually re-select the correct ROM every time. It’s like it doesn’t recognise the files as proper Amiga ROMs as such.

Thank you.

**Reply by midwan** [reply, marked as answer]
OK, then I suspect I know what might be going on.
First of all, you really should be using the newer versions with MacOS, since it has no problem running the more demanding full CPU emulation (vs the 5.x versions, which have a trimmed down CPU emulation core, meant for slower boards).

There was a bug that I recently fixed, which is probably related to this. If you had selected a custom ROM filename, instead of one from the dropdown after a Rescan, that would be "lost" when you reloaded the config. The fix is in the upcoming new version (which will be v7.x), soon to be released.

**Reply by jameswhit82** [reply]
Thats brilliant thank you, I will have a go with the preview 6.3.4 version then and let you know what happens. That definitely does sound like the issue I am experiencing though with the ROMs.

**Reply by jameswhit82** [reply]
v6.3.4 does seem to run very smoothly but I am am getting the issue with the ROM being lost  when reloading the configuration, so hopefully v7.x will resolve this for me then.

I’m guessing just renaming the ROM names to a standard format wouldn’t help at all for this issue currently?

Thank you for your help!

**Reply by midwan** [reply]
The filenames don't really matter, as it will check for the checksum values of the ROMs. If it recognizes them, it will show them with their standard names in the dropdown (you can trigger a Rescan from the GUI, by going to the Paths panel, and clicking on the relevant button there). It should show you a list of ROMs after that (more than just the AROS ones).

The bug I mentioned, is triggered when you manually select a filename to a ROM (by clicking on the button to load one up manually). That gets saved normally in the config, but it doesn't show up in the GUI when you re-load the config. That bug is fixed in v7 already, but the new release is not ready yet.

**Reply by jameswhit82** [reply]
Ok, that makes sense. Would you expect normally that the Amiga Forever ROMs wouldn’t be recognised? The ones I got were from the Amiga Forever Essentials package from Google Play, so not sure if that makes a difference at all?

No probs, that’s good news about the fix though, so will make sure I upgrade when that becomes available. Thank you.

**Reply by midwan** [reply]
Amiga Forever roms are fully supported.
Older versions were encrypted and came with a `rom.key` file, make sure you include that, if you have it.
Newer versions need to be opened once in Amiga Forever, before they are decrypted. Then they can be used without a key.

I can only imagine one of the above two scenarios is going on with your case :)

**Reply by giantclambake** [reply]
Strangely, the Amiga Forever Essentials for Android package, can't be found in in Google Play here (AU locale) ..is it geo-locked?

Anyhow, I downloaded the apk out of band to have a look ~ it contained non-standard kickstart files, and a rom.key.  

I am unsure how amiberry handles that situation, but as it stands those roms won't show up, as there's bound to fail the crc check....might this be the reason? ...ie; when the emulation is started, the rom.key is applied, but elsewise the roms exist as being encoded?

**Reply by midwan** [reply]
Yes, if they come with a `rom.key`, that needs to be included in the same directory as the rest of the ROMs. It will be used to decode them.

**Reply by jameswhit82** [reply]
That’s odd then, I do have the original Rom.key file in the same folder as the ROMs.

Very odd, I noticed the other day when I was going to try and re-extract the ROMs that I could no longer find the package on the Google Play store either. I’ll see if I can go back through my purchases and find it that way maybe.

Will keep playing around and let you know how I get on!

**Reply by midwan** [reply]
Does it work if you manually select one of those ROMs? 
If the `key.rom` is used as it should, then it should work. But then you could be running into the bug I mentioned, since the roms are not seen as "standard" due to their encryption, and your manual choice doesn't stick after you reload the config. If that's the case, it will be fixed with the new release.

If it doesn't work when you select it, then that would indicate there's an issue with the `rom.key` handling somewhere. It's been years since I tested that, since all the recent versions of AF don't use it anymore, but it's possible we have a bug.

**Reply by giantclambake** [reply]
Just did a sanity check here with `master` -- moved all existing roms out of /Amiberry/roms ---> copied rom files and rom.key from Amiga Forever Essentials for Android package into /Amiberry/roms ----> launch amiberry -> GUI -> Paths panel --> click on `Rescan Paths` button ---> goto ROM panel ....the kickstart roms are listed correctly --> click `Start` button and emulation starts normally (boots to ks1.3 insert disk screen).

Seems sane to me here in linux =)

**Reply by jameswhit82** [reply]
I’ve just done a quick check to make sure the Rom.key file is working properly, I removed it from the ‘Kickstart’ folder and I just get a blank screen when trying to start. Put it back and it’s running fine, so it’s definitely decrypting the ROMs ok.

I have noticed in the dropdown I’ve got the same ROM file listed 3 times now though:

![image](https://github.com/user-attachments/assets/e51b30c9-4153-4573-8303-ed14d12a9ff0)

Just reloaded Amiberry and my configuration and they’ve all disappeared again though:

![image](https://github.com/user-attachments/assets/2b11d7f3-edba-4f15-b815-6b04191e088e)

**Reply by jameswhit82** [reply]
Looks like Google Play has definitely removed the package for Essentials:

![image](https://github.com/user-attachments/assets/23a5f286-e474-4e37-9140-41d3f976fd51)

I’m unable to see it via my purchase history too!

**Reply by giantclambake** [reply]
It's also gone from the ChromeOS store...what's even more curious, is not a whisper from any quarter, it's as though nobody has even noticed...pretty weird. If that weren't enough, other **paid** apps on google play store, 'recommend' and 'require' the AF essentials package to work...I'll float it on EAB, and see how many versions of reality step forward ;)

Back to issue... from what you describe, it must be a macOS specific happening ; I don't have that hw .... counts me out sorry  ;^/

**Reply by jameswhit82** [reply]
Yeah that is very odd, can’t find any reason why they’ve been removed!

Interestingly I managed to boot up Amiga Forever 10 on my laptop and have managed to copy over the ROM files from there. It appears that the versions without the Rom.key file work fine on my Mac version. I now get the proper list of ROM files in the drop down :)
